src/api/routes/address.ts

```diff
@@ -107,10 +111,12 @@ interface AddressAssetEvents {
 
 export function createAddressRouter(db: DataStore, chainId: ChainID): express.Router {
   const router = express.Router();
-  const cacheHandler = getChainTipCacheHandler(db);
+  const cacheHandler = getETagCacheHandler(db);
+  const mempoolCacheHandler = getETagCacheHandler(db, ETagType.mempool);
 
   router.get(
     '/:stx_address/stx',
+    cacheHandler,
```

Comment:
Seems like there could be a significant amount of potential CACHE-OK responses marked as expired in requests that do _not_ include `unanchored=true`, and are only invalidated by the chaintip as opposed to the mempool.

We could definitely merge in this PR before addressing that, but something to keep in mind. We'd probably have to do some refactoring on how we pass around cache state.

Reply:
Great point.

I could read the query params inside the chain tip `cacheHandler` to check if it's an explicit anchored request and check the etag based on that. I'd need to add another `ETagType` like `anchored_chain_tip` that only keeps the latest `index_block_hash`. I don't think it would be that hard to do.

I'll open an issue for this.

Reply:
Opened https://github.com/hirosystems/stacks-blockchain-api/issues/1122